core/src/main/java/com/redhat/lightblue/client/Query.java

```diff
@@ -0,0 +1,245 @@
+package com.redhat.lightblue.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ContainerNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+/**
+ * A query expression
+ */
+public class Query extends Expression
+    implements Update.UpdateQuery {
+
+    public static final BinOp eq=BinOp.eq;
+    public static final BinOp neq=BinOp.neq;
+    public static final BinOp lt=BinOp.lt;
+    public static final BinOp gt=BinOp.gt;
+    public static final BinOp lte=BinOp.lte;
+    public static final BinOp gte=BinOp.gte;
+
+    public static final NaryOp in=NaryOp.in;
+    public static final NaryOp nin=NaryOp.nin;
+
+    public static final ArrOp any=ArrOp.any;
+    public static final ArrOp all=ArrOp.all;
+    public static final ArrOp non=ArrOp.none;
+
+    public static final LogOp and=LogOp.and;
+    public static final LogOp or=LogOp.or;
```

Comment:
You already have these defined as enums. It seems weird to me to then put them into constants.